backend/controller/controller.go

```diff
@@ -1340,8 +1340,8 @@ func (s *Service) executeAsyncCalls(ctx context.Context) (interval time.Duration
 	}
 
 	queueDepth := call.QueueDepth
-	didScheduleAnotherCall, err := s.dal.CompleteAsyncCall(ctx, call, callResult, func(tx *dal.DAL, isFinalResult bool) error {
```

Comment:
I think this is incorrect because if the lease is broken, the row will already be updated to a failed state. So we only want this to succeed if the context is still valid (ie the lease is still valid).